clinicaltrials.gov.js

```diff
@@ -0,0 +1,273 @@
+{
+	"translatorID": "874d70a0-6b95-4391-a681-c56dabaa1411",
+	"label": "clinicaltrials.gov",
+	"creator": "Ryan Velazquez",
+	"target": "^https://(www\\.)?clinicaltrials\\.gov/",
+	"minVersion": "3.0",
+	"maxVersion": "",
+	"priority": 100,
+	"inRepository": true,
+	"translatorType": 4,
+	"browserSupport": "gcsibv",
+	"lastUpdated": "2020-04-01 14:00:00"
+}
+
+/*
+	***** BEGIN LICENSE BLOCK *****
+
+	Copyright © 2020 Ryan Velazquez
+	
+	This file is part of Zotero.
+
+	Zotero is free software: you can redistribute it and/or modify
+	it under the terms of the GNU Affero General Public License as published by
+	the Free Software Foundation, either version 3 of the License, or
+	(at your option) any later version.
+
+	Zotero is distributed in the hope that it will be useful,
+	but WITHOUT ANY WARRANTY; without even the implied warranty of
+	MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+	GNU Affero General Public License for more details.
+
+	You should have received a copy of the GNU Affero General Public License
+	along with Zotero. If not, see <http://www.gnu.org/licenses/>.
+
+	***** END LICENSE BLOCK *****
+*/
+
+function detectWeb(doc, url) {
+	if (url.includes("https://clinicaltrials.gov/ct2/results")) {
+		throw new Error("clinicaltrials.gov search pages not supported by Zotero, only individual trials");
+	}
+	else {
+		return "report";
+	}
+}
+
+function doWeb(doc, url) {
+	scrape(doc, url);
+}
+
+function isJsonAPIRequest(url) {
+	if (url.includes("https://clinicaltrials.gov/api/query") && url.includes("fmt=JSON")) {
+		return true;
+	}
+	else {
+		return false;
+	}
+}
+
+function isXmlAPIRequest(url) {
+	if (url.includes("https://clinicaltrials.gov/api/query") && url.includes("fmt=XML")) {
+		return true;
+	}
+	else {
+		return false;
+	}
+}
+
+function getClinicalTrialID(url) {
+	if (isXmlAPIRequest(url) || isJsonAPIRequest(url)) {
+		return url.split("expr=")[1].split("&")[0];
+	}
+	else {
+		return url.split("/show/")[1].split("?")[0];
+	}
+}
+
+function dateTimeToDateString(dateTime) {
+	return dateTime.split(" ")[0].split(":").join("-");
+}
+
+function nameToFirstAndLast(rawName) {
```

Comment:
Could you add a comment on what exactly this does, ideally with a data example. If I'm reading this correctly, this would also remove authors initials/second names which doesn't seem ideal. Note that Zotero has a built-in `ZU.cleanAuthor` function that might be helpful here depending on the format.

Reply:
I felt like I was likely reinventing the wheel when I wrote that function but couldn't find a good replacement. Thanks for pointing out `ZU.cleanAuthor` that was just what I needed!